quiz/quizfinal.js

```diff
@@ -77,132 +70,207 @@ function getRandomQuestions(quizQuestionsArr) {
 
   let quantity = 0;
   while (quantity < quizQuestionsQuantity) {
-    // for (let i = 0; i < quizQuestionsArr.length; i++) {
     const randomQuestionIndex = Math.floor(
       Math.random() * quizQuestionsArr.length
     );
     randomQuiz.push(quizQuestionsArr[randomQuestionIndex]);
     quizQuestionsArr.splice(randomQuestionIndex, 1);
```

Comment:
this will change the original quizQuestionsArr. You should create a copy of it before entering the while loop and mutate that copy.